senders/mattermost/mattermost.go

```diff
@@ -0,0 +1,173 @@
+// Package mattermost is Moira sender for Mattermost - Open Source chat.
+package mattermost
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/moira-alert/moira"
+	"github.com/moira-alert/moira/senders"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+)
+
+type Sender struct {
+	frontURI string
+	location *time.Location
+	client   Client // client is interface to be possible to stub Mattermost model.Client4 dependency.
+}
+
+func NewSender() *Sender {
+	return &Sender{}
+}
+
+// Init configures Sender.
+func (sender *Sender) Init(senderSettings map[string]string, _ moira.Logger, location *time.Location, _ string) error {
+	url := senderSettings["url"]
+	if url == "" {
+		return fmt.Errorf("can not read Mattermost url from config")
+	}
+	sender.client = model.NewAPIv4Client(url)
+
+	token := senderSettings["api_token"]
+	if token == "" {
+		return fmt.Errorf("can not read Mattermost api_token from config")
+	}
+	sender.client.SetToken(token)
+
+	sender.frontURI = senderSettings["front_uri"]
```

Comment:
if uri will empty?

Reply:
Then trigger URI will be incorrect. I will fix it, thanks

Reply:
may better validate all struct

Reply:
Fixed https://github.com/moira-alert/moira/pull/791/commits/acd04f98e4599dc30cce3204c610c401a45ee418

Reply:
> may better validate all struct

Now it is ok